### PR TITLE
[SYCLomatic] Split CallExprRewriter.cpp

### PR DESCRIPTION
**Pull request by yihwang-nv**
Signed-off-by: Wang, Yihan <yihan.wang@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/CallExprRewriterCommon.h [hunk 1]
```diff
@@ -0,0 +1,1741 @@
+#ifndef DPCT_CALL_EXPR_REWRITER_COMMON_H
+#define DPCT_CALL_EXPR_REWRITER_COMMON_H
+
```
Comment (wenhuiNi) [inline clang/lib/DPCT/CallExprRewriterCommon.h:3]: Maybe , we can create CallExprRewriterCommon.cpp , and put the implementation in the source file. When the specific implementation is modified, all source include this header file would need not be recompiled.

**Comment by yihanwg** [comment]
Sorry, I'm too lazy, there are too many make*/create* functions, this draft just used to persuade folks that we can split this file and improve build speed.

**Review by wenhuiNi — APPROVED** [review]

**Review by ziranzha — APPROVED** [review]
LGTM

**Review by zhiweij1 — APPROVED** [review]

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by intwanghao — APPROVED** [review]
LGTM

File: clang/lib/DPCT/CallExprRewriter.cpp [hunk 2]
```diff
@@ -2867,297 +44,85 @@ std::function<bool(const CallExpr *C)> hasManagedAttr(int Idx) {
...
 
-class IsDefinedInCUDA {
-public:
-  IsDefinedInCUDA() {}
-  bool operator()(const CallExpr *C) {
-    auto FD = C->getDirectCallee();
-    if (!FD)
-      return false;
-    return dpct::DpctGlobalInfo::isInCudaPath(FD->getLocation());
+DerefExpr DerefExpr::create(const Expr *E, const CallExpr * C) {
```
Comment (tangjj11) [inline clang/lib/DPCT/CallExprRewriter.cpp:47]: not format here.

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by ShengchenJ — APPROVED** [review]

**Review by tangjj11 — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**